execute/row_fn.go

```diff
@@ -44,35 +54,45 @@ func (f *dynamicFn) typeof(cols []flux.ColMeta) (semantic.MonoType, error) {
 }
 
 func (f *dynamicFn) prepare(cols []flux.ColMeta, extraTypes map[string]semantic.MonoType) (preparedFn, error) {
-	// Prepare the type of the record column.
-	recordType, err := f.typeof(cols)
-	if err != nil {
-		return preparedFn{}, err
-	}
+	// If the types have not changed we do not need to recompile, just use the cached version
+	if f.compiledFn == nil || !reflect.DeepEqual(f.compiledFn.cols, cols) || !reflect.DeepEqual(f.compiledFn.extraTypes, extraTypes) {
```

Comment:
Should we implement an explicit equality measure for these values instead of using reflect.DeepEqual?

Reply:
Didn't want to write it out in case this didn't work. fixed.